File: frontend/src/api/bot/process.ts
import { request } from '../base'

export interface PortInfo {
  protocol: string
  local: string
  remote: string
  state: string
}

export interface ProcessPortInfo {
  pid: number
  ports: PortInfo[]
  error?: string
}

// 获取进程占用的端口信息
export const apiGetProcessPorts = async (
  pid: number
): Promise<ProcessPortInfo> => {
  return new Promise((resolve, reject) => {
    request({
      url: `/bot/process/${pid}/ports`,
      method: 'GET'
    })
      .then(res => res.data)
      .then(resolve)
      .catch(reject)
  })
}

File: src/models/process.go
package models

// ProcessPortInfo 进程端口信息
type ProcessPortInfo struct {
	PID   int        `json:"pid"`             // 进程ID
	Ports []PortInfo `json:"ports"`           // 端口列表
	Error string     `json:"error,omitempty"` // 错误信息
}

// PortInfo 端口信息
type PortInfo struct {
	Protocol string `json:"protocol"` // 协议类型 (TCP/UDP)
	Local    string `json:"local"`    // 本地地址:端口
	Remote   string `json:"remote"`   // 远程地址:端口
	State    string `json:"state"`    // 连接状态
	PID      string `json:"pid"`      // 进程ID
}
